Logger/BaseEntity.cs

```diff
@@ -0,0 +1,7 @@
+namespace Logger;
+
+public abstract class BaseEntity : IEntity
```

Comment:
```suggestion
public abstract record BaseEntity : IEntity
```
You'll want to define this as an abstract **_record_** class, that way you can inherit from it on your book employee and student records.